app/models/edition/custom_lead_image.rb

```diff
@@ -37,4 +34,20 @@ def remove_lead_image
 
     edition_lead_image.destroy!
   end
+
+  def body_does_not_contain_lead_image_markdown
+    return if lead_image.blank?
+
+    if body_contains_lead_image_filename_markdown? || body_contains_lead_image_index_markdown?
+      errors.add(:body, "cannot have a reference to the lead image in the text")
+    end
+  end
+
+  def body_contains_lead_image_filename_markdown?
+    body.match?(/\[Image:\s*#{Regexp.escape(lead_image.filename)}\s*\]/)
+  end
+
+  def body_contains_lead_image_index_markdown?
+    body.match?(/^!!#{lead_image_index + 1}([^\d]|$)/)
```

Comment:
Could we try and see if this passes?

```suggestion
    body.match?(/!!#{lead_image_index + 1}[^\d]/)
```

Reply:
this doesn't work unfortunately. After some discussion we've decided to go as is